website: add links to schema references

kensternberg-authentik opened the pull request:
## Details

website: add links to schema references

### What:

Added links to the schema references on the blueprints page, and a new collapsible section that shows all the past schemas as referenced in the original edit.

### Why:

URLs to live objects (URLs that aren’t just examples) should be live links themselves. Referring to past versions of the documents and not providing a list of those versions or links to them seemed incomplete.

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

By default, the "Collapse" section is closed when you first reach the page:

![image](https://github.com/user-attachments/assets/5c8ddbd0-41de-4d16-965b-ec266f8ea843)

When the button is clicked, it opens to reveal this:

![image](https://github.com/user-attachments/assets/d4bc3c51-7598-4816-8b47-61dc6ab2989d)

kensternberg-authentik committed:
web: fix esbuild issue with style sheets

Getting ESBuild, Lit, and Storybook to all agree on how to read and parse stylesheets is a serious
pain. This fix better identifies the value types (instances) being passed from various sources in
the repo to the three *different* kinds of style processors we're using (the native one, the
polyfill one, and whatever the heck Storybook does internally).

Falling back to using older CSS instantiating techniques one era at a time seems to do the trick.
It's ugly, but in the face of the aggressive styling we use to avoid Flashes of Unstyled Content
(FLoUC), it's the logic with which we're left.

In standard mode, the following warning appears on the console when running a Flow:

```
Autofocus processing was blocked because a document already has a focused element.
```

In compatibility mode, the following **error** appears on the console when running a Flow:

```
crawler-inject.js:1106 Uncaught TypeError: Failed to execute 'observe' on 'MutationObserver': parameter 1 is not of type 'Node'.
    at initDomMutationObservers (crawler-inject.js:1106:18)
    at crawler-inject.js:1114:24
    at Array.forEach (<anonymous>)
    at initDomMutationObservers (crawler-inject.js:1114:10)
    at crawler-inject.js:1549:1
initDomMutationObservers @ crawler-inject.js:1106
(anonymous) @ crawler-inject.js:1114
initDomMutationObservers @ crawler-inject.js:1114
(anonymous) @ crawler-inject.js:1549
```

Despite this error, nothing seems to be broken and flows work as anticipated.

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  tenants: really ensure default tenant cannot be deleted (#8875)
  core: bump github.com/go-openapi/runtime from 0.27.2 to 0.28.0 (#8867)
  core: bump pytest from 8.0.2 to 8.1.1 (#8868)
  core: bump github.com/go-openapi/strfmt from 0.22.2 to 0.23.0 (#8869)
  core: bump bandit from 1.7.7 to 1.7.8 (#8870)
  core: bump packaging from 23.2 to 24.0 (#8871)
  core: bump ruff from 0.3.1 to 0.3.2 (#8873)
  web: bump the wdio group in /tests/wdio with 3 updates (#8865)
  core: bump requests-oauthlib from 1.3.1 to 1.4.0 (#8866)
  core: bump uvicorn from 0.27.1 to 0.28.0 (#8872)
  core: bump django-filter from 23.5 to 24.1 (#8874)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: fix esbuild issue with style sheets (#8856)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: upgrade to lit 3 (#8781)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  Update _envoy_istio.md (#8888)
  website/docs: new landing page for Providers (#8879)
  web: bump the sentry group in /web with 1 update (#8881)
  web: bump chromedriver from 122.0.4 to 122.0.5 in /tests/wdio (#8884)
  web: bump the eslint group in /tests/wdio with 2 updates (#8883)
  web: bump the eslint group in /web with 2 updates (#8885)
  website: bump @types/react from 18.2.64 to 18.2.65 in /website (#8886)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  api: capabilities: properly set can_save_media when s3 is enabled (#8896)
  web: bump the rollup group in /web with 3 updates (#8891)
  core: bump pydantic from 2.6.3 to 2.6.4 (#8892)
  core: bump twilio from 9.0.0 to 9.0.1 (#8893)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump chromedriver from 122.0.5 to 122.0.6 in /tests/wdio (#8902)
  web: bump vite-tsconfig-paths from 4.3.1 to 4.3.2 in /web (#8903)
  core: bump google.golang.org/protobuf from 1.32.0 to 1.33.0 (#8901)
  web: provide InstallID on EnterpriseListPage (#8898)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: clean up and remove redundant alias '@goauthentik/app' (#8889)
  web/admin: fix markdown table rendering (#8908)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (31 commits)
  root: support redis username (#8935)
  core: bump black from 24.2.0 to 24.3.0 (#8945)
  web: bump the wdio group in /tests/wdio with 2 updates (#8939)
  web: bump the sentry group in /web with 1 update (#8941)
  website: bump postcss from 8.4.35 to 8.4.36 in /website (#8940)
  core: bump twilio from 9.0.1 to 9.0.2 (#8942)
  core: bump ruff from 0.3.2 to 0.3.3 (#8943)
  events: discard notification if user has empty email (#8938)
  ci: always run ci-main on branch pushes (#8950)
  core: bump goauthentik.io/api/v3 from 3.2024022.2 to 3.2024022.3 (#8946)
  website/docs: add new name "Microsft Entra ID" for Azure AD  (#8930)
  outposts: Enhance config options for k8s outposts (#7363)
  website/docs: add link to CRUD docs (#8925)
  web: bump API Client version (#8927)
  outpost: improved set secret answers for flow execution (#8013)
  stages/user_write: ensure user data is json-serializable (#8926)
  website/docs: update example ldapsearch commands (#8906)
  admin: Handle latest  version unknown in admin dashboard (#8858)
  core: bump coverage from 7.4.3 to 7.4.4 (#8917)
  core: bump urllib3 from 1.26.18 to 2.2.1 (#8918)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  outposts/proxy: Fix invalid redirect on external hosts containing path components (#8915)
  core: cache user application list under policies (#8895)
  web: bump the eslint group in /web with 2 updates (#8959)
  web: bump core-js from 3.36.0 to 3.36.1 in /web (#8960)
  website: bump @types/react from 18.2.66 to 18.2.67 in /website (#8962)
  web: bump the eslint group in /tests/wdio with 2 updates (#8963)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: improve build speeds even moar!!!!!! (#8954)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: config: remove options moved to tenants (#8976)
  web: bump @types/grecaptcha from 3.0.8 to 3.0.9 in /web (#8971)
  web: bump country-flag-icons from 1.5.9 to 1.5.10 in /web (#8970)
  web: bump the babel group in /web with 7 updates (#8969)
  core: bump uvicorn from 0.28.0 to 0.28.1 (#8968)
  website: bump postcss from 8.4.36 to 8.4.37 in /website (#8967)
  internal: cleanup static file serving setup code (#8965)
  website/integrations: portainer: match portainer settings order (#8974)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add example policy to enforce unique email address (#8955)
  web/admin: remove enterprise preview banner (#8991)
  core: bump uvicorn from 0.28.1 to 0.29.0 (#8980)
  core: bump sentry-sdk from 1.42.0 to 1.43.0 (#8981)
  web: bump the babel group in /web with 3 updates (#8983)
  web: bump typescript from 5.4.2 to 5.4.3 in /web (#8984)
  web: bump typescript from 5.4.2 to 5.4.3 in /tests/wdio (#8986)
  web: bump chromedriver from 122.0.6 to 123.0.0 in /tests/wdio (#8987)
  website: bump typescript from 5.4.2 to 5.4.3 in /website (#8989)
  core: bump importlib-metadata from 7.0.2 to 7.1.0 (#8982)
  web: bump the wdio group in /tests/wdio with 3 updates (#8985)
  website: bump postcss from 8.4.37 to 8.4.38 in /website (#8988)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#9021)
  sources/ldap: add ability to disable password write on login (#8377)
  web: bump API Client version (#9020)
  lifecycle: migrate: ensure template schema exists before migrating (#8952)
  website/integrations: Update nextcloud Admin Group Expression (#7314)
  web/flow: general ux improvements (#8558)
  website: bump @types/react from 18.2.67 to 18.2.69 in /website (#9016)
  core: bump requests-oauthlib from 1.4.0 to 2.0.0 (#9018)
  web: bump the sentry group in /web with 2 updates (#9017)
  web/admin: small fixes (#9002)
  website: bump webpack-dev-middleware from 5.3.3 to 5.3.4 in /website (#9001)
  core: bump ruff from 0.3.3 to 0.3.4 (#8998)
  website/docs: Upgrade nginx reverse porxy config (#8947)
  website/docs: improve flow inspector docs (#8993)
  website/deverlop-docs website/integrations: add links to integrations template (#8995)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#9035)
  website/docs: maintenance, re-add system settings (#9026)
  core: bump duo-client from 5.2.0 to 5.3.0 (#9029)
  website: bump express from 4.18.2 to 4.19.2 in /website (#9027)
  web: bump express from 4.18.3 to 4.19.2 in /web (#9028)
  web: bump the eslint group in /web with 2 updates (#9030)
  core: bump goauthentik.io/api/v3 from 3.2024022.3 to 3.2024022.5 (#9031)
  website: bump @types/react from 18.2.69 to 18.2.70 in /website (#9032)
  web: bump the eslint group in /tests/wdio with 2 updates (#9033)
  web: bump katex from 0.16.9 to 0.16.10 in /web (#9025)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in fr (#9023)
  website/docs: include OS-specific docker-compose install instructions + minor fixes (#8975)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: a few minor bugfixes and lintfixes (#9044)
  website/integrations: add documentation for OIDC setup with Xen Orchestra (#9000)
  website: bump @types/react from 18.2.70 to 18.2.72 in /website (#9041)
  core: bump goauthentik.io/api/v3 from 3.2024022.5 to 3.2024022.6 (#9042)
  web: fix markdown rendering bug for alerts (#9037)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (23 commits)
  providers/oauth2: fix interactive device flow (#9076)
  website/docs: fix transports example (#9074)
  events: fix log_capture (#9075)
  web: bump the sentry group in /web with 2 updates (#9065)
  core: bump goauthentik.io/api/v3 from 3.2024022.6 to 3.2024022.7 (#9064)
  web: bump @codemirror/lang-python from 6.1.4 to 6.1.5 in /web (#9068)
  web: bump the eslint group in /web with 1 update (#9066)
  web: bump glob from 10.3.10 to 10.3.12 in /web (#9069)
  web: bump the rollup group in /web with 3 updates (#9067)
  web: bump the eslint group in /tests/wdio with 1 update (#9071)
  core: bump webauthn from 2.0.0 to 2.1.0 (#9070)
  core: bump sentry-sdk from 1.43.0 to 1.44.0 (#9073)
  core: bump requests-mock from 1.12.0 to 1.12.1 (#9072)
  web: bump API Client version (#9061)
  events: rework log messages returned from API and their rendering (#8770)
  website/docs: update airgapped config (#9049)
  website: bump @types/react from 18.2.72 to 18.2.73 in /website (#9052)
  web: bump the rollup group in /web with 3 updates (#9053)
  core: bump django-filter from 24.1 to 24.2 (#9055)
  core: bump requests-mock from 1.11.0 to 1.12.0 (#9056)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: move context controllers into reactive controller plugins (#8996)
  web: maintenance: split tsconfig into “base” and “build” variants. (#9036)
  web: consistent style declarations internally (#9077)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump @patternfly/elements from 2.4.0 to 3.0.0 in /web (#9089)
  web: bump ts-pattern from 5.0.8 to 5.1.0 in /web (#9090)
  website: bump the docusaurus group in /website with 9 updates (#9087)
  web/admin: allow custom sorting for bound* tables (#9080)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  root: fix missing imports after #9081 (#9106)
  root: move database calls from ready() to dedicated startup signal (#9081)
  web: fix console log leftover (#9096)
  web: bump the eslint group in /web with 2 updates (#9098)
  core: bump twilio from 9.0.2 to 9.0.3 (#9103)
  web: bump the eslint group in /tests/wdio with 2 updates (#9099)
  core: bump drf-spectacular from 0.27.1 to 0.27.2 (#9100)
  core: bump django-model-utils from 4.4.0 to 4.5.0 (#9101)
  core: bump ruff from 0.3.4 to 0.3.5 (#9102)
  website/docs:  update notes on SECRET_KEY (#9091)
  web: fix broken locale compile (#9095)
  website/integrations: add outline knowledge base (#8786)
  website/docs: fix typo (#9082)
  website/docs: email stage: fix example translation error (#9048)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (22 commits)
  blueprints: fix default username field in user-settings flow (#9136)
  website/docs: add procedural docs for RAC (#9006)
  web: bump API Client version (#9133)
  ci: fix python client generator (#9134)
  root: generate python client (#9107)
  web: Bump vite from 5.1.4 to 5.2.8 in /web (#9120)
  core, web: update translations (#9124)
  core: Bump golang from 1.22.1-bookworm to 1.22.2-bookworm (#9125)
  web: Bump the babel group in /web with 2 updates (#9126)
  web: Bump the eslint group in /web with 1 update (#9127)
  web: Bump the eslint group in /tests/wdio with 1 update (#9129)
  core: Bump sentry-sdk from 1.44.0 to 1.44.1 (#9130)
  core: Bump channels from 4.0.0 to 4.1.0 (#9131)
  core: Bump django from 5.0.3 to 5.0.4 (#9132)
  web: Bump the rollup group in /web with 3 updates (#9128)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9110)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9109)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9111)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9112)
  web: Bump @fortawesome/fontawesome-free from 6.5.1 to 6.5.2 in /web (#9116)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (25 commits)
  root: fix readme (#9178)
  enterprise: fix audit middleware import (#9177)
  web: bump @spotlightjs/spotlight from 1.2.16 to 1.2.17 in /web in the sentry group (#9162)
  web: bump API Client version (#9174)
  stages/authenticator_webauthn: add MDS support (#9114)
  website/integrations: Update Nextcloud OIDC secret size limitation (#9139)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9170)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9171)
  web: bump the rollup group in /web with 3 updates (#9164)
  web: bump @codemirror/legacy-modes from 6.3.3 to 6.4.0 in /web (#9166)
  web: bump ts-pattern from 5.1.0 to 5.1.1 in /web (#9167)
  core: bump github.com/go-ldap/ldap/v3 from 3.4.6 to 3.4.7 (#9168)
  core, web: update translations (#9156)
  root: fix redis username in lifecycle (#9158)
  web: ak-checkbox-group for short, static, multi-select events (#9138)
  root: fix startup (#9151)
  core: Bump golang.org/x/oauth2 from 0.18.0 to 0.19.0 (#9146)
  core: Bump twilio from 9.0.3 to 9.0.4 (#9143)
  web: Bump country-flag-icons from 1.5.10 to 1.5.11 in /web (#9144)
  web: Bump typescript from 5.4.3 to 5.4.4 in /web (#9145)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web/admin: fix SAML Provider preview (#9192)
  core, web: update translations (#9183)
  web: bump chromedriver from 123.0.1 to 123.0.2 in /tests/wdio (#9188)
  website: bump @types/react from 18.2.74 to 18.2.75 in /website (#9185)
  website/docs: update Postgresql username (#9190)
  core: bump maxmind/geoipupdate from v6.1 to v7.0 (#9186)
  events: add context manager to ignore/modify audit events being written (#9181)
  web: fix application library list display length and capability (#9094)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9194)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9197)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9196)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9198)
  web: preserve selected list when provider updates (#9200)
  web: bump API Client version (#9195)
  sources/oauth: make URLs not required, only check when no OIDC URLs are defined (#9182)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add note for flow compatibility mode (#9204)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add more info and links about enforciing unique email addresses (#9154)
  core: bump goauthentik.io/api/v3 from 3.2024022.7 to 3.2024022.8 (#9215)
  web: bump API Client version (#9214)
  stages/authenticator_validate: add ability to limit webauthn device types (#9180)
  web: bump API Client version (#9213)
  core: add user settable token durations (#7410)
  core, web: update translations (#9205)
  web: bump typescript from 5.4.4 to 5.4.5 in /tests/wdio (#9206)
  web: bump chromedriver from 123.0.2 to 123.0.3 in /tests/wdio (#9207)
  core: bump sentry-sdk from 1.44.1 to 1.45.0 (#9208)
  web: bump typescript from 5.4.4 to 5.4.5 in /web (#9209)
  website: bump typescript from 5.4.4 to 5.4.5 in /website (#9210)
  core: bump python from 3.12.2-slim-bookworm to 3.12.3-slim-bookworm (#9211)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (21 commits)
  web: manage stacked modals with a stack (#9193)
  website/docs: ensure yaml code blocks have language tags (#9240)
  blueprints: only create default brand if no other default brand exists (#9222)
  web: bump API Client version (#9239)
  website/integrations: portainer: Fix Redirect URL mismatch (#9226)
  api: fix authentication schema (#9238)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9229)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9230)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9228)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9231)
  core: bump pydantic from 2.6.4 to 2.7.0 (#9232)
  core: bump ruff from 0.3.5 to 0.3.7 (#9233)
  web: bump @sentry/browser from 7.109.0 to 7.110.0 in /web in the sentry group (#9234)
  website: bump @types/react from 18.2.75 to 18.2.77 in /website (#9236)
  core, web: update translations (#9225)
  website/integrations: add pfSense search scope (#9221)
  core: bump idna from 3.6 to 3.7 (#9224)
  website/docs: add websocket support to nginx snippets (#9220)
  internal: add tests to go flow executor (#9219)
  website/integrations: nextcloud: add tip to solve hashed groups configuring OAuth2 (#9153)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (34 commits)
  web: bump API Client version (#9299)
  core: fix api schema for users and groups (#9298)
  providers/oauth2: fix refresh_token grant returning incorrect id_token (#9275)
  web: bump @sentry/browser from 7.110.0 to 7.110.1 in /web in the sentry group (#9278)
  core, web: update translations (#9277)
  web: bump the rollup group in /web with 3 updates (#9280)
  web: bump lit from 3.1.2 to 3.1.3 in /web (#9282)
  web: bump @lit/context from 1.1.0 to 1.1.1 in /web (#9281)
  website: bump @types/react from 18.2.78 to 18.2.79 in /website (#9286)
  core: bump goauthentik.io/api/v3 from 3.2024022.10 to 3.2024022.11 (#9285)
  core: bump sqlparse from 0.4.4 to 0.5.0 (#9276)
  lifecycle: gunicorn: fix app preload (#9274)
  events: add indexes (#9272)
  web/flows: fix passwordless hidden without input (#9273)
  root: fix geoipupdate arguments (#9271)
  website/docs: cleanup more (#9249)
  web: bump API Client version (#9270)
  sources: add SCIM source (#3051)
  core: delegated group member management (#9254)
  web: bump API Client version (#9269)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (23 commits)
  web: bump API Client version (#9316)
  release: 2024.2.3
  website/docs: 2024.2.3 release notes (#9313)
  web/admin: fix log viewer empty state (#9315)
  website/docs: fix formatting for stage changes (#9314)
  core: bump github.com/go-ldap/ldap/v3 from 3.4.7 to 3.4.8 (#9310)
  core: bump goauthentik.io/api/v3 from 3.2024022.11 to 3.2024022.12 (#9311)
  web: bump core-js from 3.36.1 to 3.37.0 in /web (#9309)
  core: bump gunicorn from 21.2.0 to 22.0.0 (#9308)
  core, web: update translations (#9307)
  website/docs: system settings: add default token duration and length (#9306)
  web/flows: update flow background (#9305)
  web: fix locale loading being skipped (#9301)
  translate: Updates for file web/xliff/en.xlf in fr (#9304)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in fr (#9303)
  core: replace authentik_signals_ignored_fields with audit_ignore (#9291)
  web/flow: fix form input rendering issue (#9297)
  events: fix incorrect user logged when using API token authentication (#9302)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9293)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9295)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  stages/prompt: fix username field throwing error with existing user (#9342)
  root: expose session storage configuration (#9337)
  website/integrations: fix typo (#9340)
  root: fix go.mod for codeql checking (#9338)
  root: make redis settings more consistent (#9335)
  web/admin: fix error in admin interface due to un-hydrated context (#9336)
  web: bump API Client version (#9334)
  stages/authenticator_webauthn: fix attestation value (#9333)
  website/docs: fix SECRET_KEY length (#9328)
  website/docs: fix email template formatting (#9330)
  core, web: update translations (#9323)
  web: bump @patternfly/elements from 3.0.0 to 3.0.1 in /web (#9324)
  core: bump celery from 5.3.6 to 5.4.0 (#9325)
  core: bump goauthentik.io/api/v3 from 3.2024022.12 to 3.2024023.1 (#9327)
  sources/scim: service account should be internal (#9321)
  web: bump the storybook group in /web with 8 updates (#9266)
  sources/scim: cleanup service account when source is deleted (#9319)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (24 commits)
  web: bump the wdio group in /tests/wdio with 4 updates (#9374)
  web: bump the rollup group in /web with 3 updates (#9371)
  core: bump ruff from 0.4.0 to 0.4.1 (#9372)
  core, web: update translations (#9366)
  web/admin: fix document title for admin interface (#9362)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9363)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9364)
  core, web: update translations (#9360)
  website/docs: release notes 2024.4: add performance improvements values (#9356)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9317)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9318)
  website/docs: 2024.4 release notes (#9267)
  sources/ldap: fix default blueprint for mapping user DN to path (#9355)
  web/admin: group form dual select (#9354)
  core: bump golang.org/x/net from 0.22.0 to 0.23.0 (#9351)
  core: bump goauthentik.io/api/v3 from 3.2024023.1 to 3.2024023.2 (#9345)
  web: bump chromedriver from 123.0.3 to 123.0.4 in /tests/wdio (#9348)
  core: bump twilio from 9.0.4 to 9.0.5 (#9346)
  core: bump ruff from 0.3.7 to 0.4.0 (#9347)
  web: bump @sentry/browser from 7.110.1 to 7.111.0 in /web in the sentry group (#9349)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#9400)
  release: 2024.4.0
  release: 2024.4.0-rc1
  root: bump blueprint schema version
  lifecycle: fix ak test-all command
  website/docs: finalize 2024.4 release notes (#9396)
  web: bump @sentry/browser from 7.111.0 to 7.112.1 in /web in the sentry group (#9387)
  web: bump the rollup group in /web with 3 updates (#9388)
  ci: bump helm/kind-action from 1.9.0 to 1.10.0 (#9389)
  website: bump clsx from 2.1.0 to 2.1.1 in /website (#9390)
  core: bump pydantic from 2.7.0 to 2.7.1 (#9391)
  core: bump freezegun from 1.4.0 to 1.5.0 (#9393)
  core: bump coverage from 7.4.4 to 7.5.0 (#9392)
  web: bump the storybook group in /web with 7 updates (#9380)
  web: bump the rollup group in /web with 3 updates (#9381)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: clean up some repetitive types (#9241)
  core: fix logic for token expiration (#9426)
  ci: fix ci pipeline (#9427)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9424)
  web: Add resolved and integrity fields back to package-lock.json (#9419)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9407)
  stages/identification: don't check source component (#9410)
  core: bump selenium from 4.19.0 to 4.20.0 (#9411)
  core: bump black from 24.4.0 to 24.4.1 (#9412)
  ci: bump golangci/golangci-lint-action from 4 to 5 (#9413)
  core: bump goauthentik.io/api/v3 from 3.2024023.2 to 3.2024040.1 (#9414)
  web: bump @sentry/browser from 7.112.1 to 7.112.2 in /web in the sentry group (#9416)
  sources/oauth: ensure all UI sources return a valid source (#9401)
  web: markdown: display markdown even when frontmatter is missing (#9404)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (43 commits)
  stages/authenticator_webauthn: Update FIDO MDS3 & Passkey aaguid blobs (#9535)
  web: bump the rollup group across 1 directory with 3 updates (#9532)
  website/developer-docs: Add note for custom YAML tags in an IDE (#9528)
  lifecycle: close database connection after migrating (#9516)
  web: bump the babel group in /web with 3 updates (#9520)
  core: bump node from 21 to 22 (#9521)
  web: bump @codemirror/lang-python from 6.1.5 to 6.1.6 in /web (#9523)
  providers/rac: bump guacd to 1.5.5 (#9514)
  core: only prefetch related objects when required (#9476)
  website/integrations: move Fortimanager to Networking (#9505)
  website: bump react-tooltip from 5.26.3 to 5.26.4 in /website (#9494)
  web: bump the rollup group in /web with 3 updates (#9497)
  web: bump yaml from 2.4.1 to 2.4.2 in /web (#9499)
  core: bump goauthentik.io/api/v3 from 3.2024040.1 to 3.2024041.1 (#9503)
  core: bump pytest from 8.1.1 to 8.2.0 (#9501)
  website: bump react-dom from 18.3.0 to 18.3.1 in /website (#9495)
  website: bump react and @types/react in /website (#9496)
  web: bump react-dom from 18.3.0 to 18.3.1 in /web (#9498)
  core: bump sentry-sdk from 2.0.0 to 2.0.1 (#9502)
  web/flows: fix missing fallback for flow logo (#9487)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web/flows: fix error when enrolling multiple WebAuthn devices consecutively (#9545)
  web: bump ejs from 3.1.9 to 3.1.10 in /tests/wdio (#9542)
  web: bump API Client version (#9543)
  providers/saml: fix ecdsa support (#9537)
  website/integrations: nextcloud: connect to existing user (#9155)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: fix openssl rand commands (#9554)
  web: bump @sentry/browser from 7.112.2 to 7.113.0 in /web in the sentry group (#9549)
  core, web: update translations (#9548)
  core: bump goauthentik.io/api/v3 from 3.2024041.1 to 3.2024041.2 (#9551)
  core: bump django-model-utils from 4.5.0 to 4.5.1 (#9550)
  providers/scim: fix time_limit not set correctly (#9546)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (42 commits)
  website/docs: prepare 2024.4.2 release notes (#9555)
  web: bump the esbuild group in /web with 2 updates (#9616)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9611)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9560)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9563)
  ci: bump golangci/golangci-lint-action from 5 to 6 (#9615)
  web: bump esbuild from 0.20.2 to 0.21.0 in /web (#9617)
  core: bump cryptography from 42.0.6 to 42.0.7 (#9618)
  core: bump sentry-sdk from 2.0.1 to 2.1.1 (#9619)
  core: bump django from 5.0.4 to 5.0.5 (#9620)
  core, web: update translations (#9613)
  website/docs: move Sources from Integrations into Docs (#9515)
  website/docs: add procedurals to flow inspector docs (#9556)
  core: bump jinja2 from 3.1.3 to 3.1.4 (#9610)
  web: clean up the options rendering in PromptForm (#9564)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9608)
  website/docs: add instructions for deploying radius manually with docker compose (#9605)
  sources/scim: fix duplicate groups and invalid schema (#9466)
  core: fix condition in task clean_expiring_models (#9603)
  translate: Updates for file web/xliff/en.xlf in fr (#9600)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  core, web: update translations (#9633)
  core: bump google-api-python-client from 2.127.0 to 2.128.0 (#9641)
  core: bump goauthentik.io/api/v3 from 3.2024041.3 to 3.2024042.2 (#9635)
  core: bump golang from 1.22.2-bookworm to 1.22.3-bookworm (#9636)
  web: bump the esbuild group in /web with 2 updates (#9637)
  web: bump esbuild from 0.21.0 to 0.21.1 in /web (#9639)
  core: bump django from 5.0.5 to 5.0.6 (#9640)
  web: bump API Client version (#9630)
  enterprise/providers/google: initial account sync to google workspace (#9384)
  web/flows: fix error when using consecutive webauthn validator stages (#9629)
  web: bump API Client version (#9626)
  website/docs: refine intro page for sources (#9625)
  release: 2024.4.2

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#9656)
  enterprise/providers/microsoft_entra: initial account sync to microsoft entra (#9632)
  web: bump chromedriver from 124.0.1 to 124.0.2 in /tests/wdio (#9652)
  web: bump @sentry/browser from 7.113.0 to 7.114.0 in /web in the sentry group (#9653)
  core, web: update translations (#9650)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9644)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9646)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9645)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9647)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add hardening advice and link directly to Cure53 results (#9670)
  core: bump goauthentik.io/api/v3 from 3.2024042.2 to 3.2024042.4 (#9674)
  core: bump ruff from 0.4.3 to 0.4.4 (#9677)
  core: bump github.com/prometheus/client_golang from 1.19.0 to 1.19.1 (#9675)
  web: bump glob from 10.3.12 to 10.3.14 in /web (#9676)
  lib/providers/sync: multiple minor fixes (#9667)
  core: fix source flow_manager not always appending save stage (#9659)
  web: bump API Client version (#9660)
  web/admin: only show non-backchannel providers in application provider select (#9658)
  website/docs: add new doc about extra steps for hardening authentik (#9649)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump esbuild from 0.21.1 to 0.21.2 in /web (#9696)
  web: bump chromedriver from 124.0.2 to 124.0.3 in /tests/wdio (#9692)
  website: bump @types/react from 18.3.1 to 18.3.2 in /website (#9691)
  core, web: update translations (#9672)
  core: bump psycopg from 3.1.18 to 3.1.19 (#9698)
  core: bump google-api-python-client from 2.128.0 to 2.129.0 (#9694)
  web: bump the esbuild group in /web with 2 updates (#9695)
  web: bump glob from 10.3.14 to 10.3.15 in /web (#9697)
  core: bump freezegun from 1.5.0 to 1.5.1 (#9693)
  web: fix value handling inside controlled components (#9648)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: update traefik to latest version in proxy provider (#9707)
  sources/saml: fix FlowPlanner error due to pickle (#9708)
  website/docs: add docs about Google Workspace (#9669)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9702)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9703)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9705)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9706)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump @sentry/browser from 7.114.0 to 8.2.1 in /web in the sentry group across 1 directory (#9757)
  core, web: update translations (#9714)
  core: bump sentry-sdk from 2.1.1 to 2.2.0 (#9753)
  core: bump selenium from 4.20.0 to 4.21.0 (#9754)
  core: bump msgraph-sdk from 1.2.0 to 1.4.0 (#9755)
  core: bump github.com/sethvargo/go-envconfig from 1.0.1 to 1.0.2 (#9756)
  web: bump chromedriver from 124.0.3 to 125.0.0 in /tests/wdio (#9758)
  website/docs: new PR for the Entra provider docs (ignore old one) (#9741)
  root: include task_id in events and logs (#9749)
  web: bump the esbuild group in /web with 2 updates (#9745)
  web: bump esbuild from 0.21.2 to 0.21.3 in /web (#9746)
  web: bump the storybook group across 1 directory with 7 updates (#9747)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (36 commits)
  ci: use container registry for container build cache (#9809)
  core: bump lxml from 5.2.1 to 5.2.2 (#9717)
  web: bump mermaid from 10.9.0 to 10.9.1 in /web (#9734)
  core: bump scim2-filter-parser from 0.5.0 to 0.5.1 (#9730)
  web: bump core-js from 3.37.0 to 3.37.1 in /web (#9733)
  stages/authenticator_webauthn: Update FIDO MDS3 & Passkey aaguid blobs (#9729)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9802)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9803)
  core: bump sentry-sdk from 2.2.0 to 2.2.1 (#9807)
  web: bump the storybook group in /web with 7 updates (#9804)
  web: bump glob from 10.3.15 to 10.3.16 in /web (#9805)
  root: docker-compose: remove version top level element (#9631)
  core, web: update translations (#9790)
  web: bump API Client version (#9801)
  web/admin: rework initial wizard pages and add grid layout (#9668)
  website/integrations: discord: fix typo (#9800)
  website/integration/netbox: fix group custom pipeline example (#9738)
  root: add primary-replica db router (#9479)
  website/integrations: add three more policy-expressions to discord-docs (#5760)
  website/integrations: netbox: add missing scope configuration (#9491)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (43 commits)
  website/docs: improve read replica docs (#9828)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9859)
  web: bump API Client version (#9857)
  lib/sync/outgoing, sources/ldap: fix sync status endpoint (#9855)
  lifecycle/migrate: only acquire lock once (#9856)
  core: add option to select group for property mapping testing (#9834)
  policies: fix ak_call_policy failing when used in testing (#9853)
  website/integrations: gitlab: make placeholder clearer (#9838)
  core: bump requests from 2.31.0 to 2.32.2 (#9852)
  core: bump codespell from 2.2.6 to 2.3.0 (#9842)
  core: bump docker from 7.0.0 to 7.1.0 (#9843)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#9839)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#9840)
  core: bump watchdog from 4.0.0 to 4.0.1 (#9845)
  website: bump @types/react from 18.3.2 to 18.3.3 in /website (#9841)
  core: bump sentry-sdk from 2.2.1 to 2.3.1 (#9844)
  core: bump goauthentik.io/api/v3 from 3.2024042.4 to 3.2024042.7 (#9846)
  web: bump chromedriver from 125.0.1 to 125.0.2 in /tests/wdio (#9847)
  web: bump @sentry/browser from 8.3.0 to 8.4.0 in /web in the sentry group (#9848)
  web: bump the storybook group in /web with 7 updates (#9849)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add docs for configuring Facebook as a source (#9884)
  web: bump the esbuild group in /web with 2 updates (#9879)
  core: bump twilio from 9.0.5 to 9.1.0 (#9875)
  core: bump coverage from 7.5.1 to 7.5.2 (#9876)
  core: bump goauthentik.io/api/v3 from 3.2024042.7 to 3.2024042.8 (#9877)
  web: bump the babel group in /web with 7 updates (#9878)
  web: bump glob from 10.4.0 to 10.4.1 in /web (#9881)
  web: bump the wdio group in /tests/wdio with 4 updates (#9882)
  events: fix geoip enrich context not converting to json-seriazable data (#9885)
  stages/user_login: fix non-json-serializable being used (#9886)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (44 commits)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#9943)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#9944)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#9942)
  stages/authenticator_webauthn: migrate device type import to systemtask and schedule (#9958)
  website: bump prettier from 3.2.5 to 3.3.0 in /website (#9951)
  web: bump yaml from 2.4.2 to 2.4.3 in /web (#9945)
  web: bump prettier from 3.2.5 to 3.3.0 in /web (#9946)
  core: bump uvicorn from 0.30.0 to 0.30.1 (#9948)
  core: bump ruff from 0.4.6 to 0.4.7 (#9949)
  website: bump the docusaurus group in /website with 9 updates (#9950)
  web: bump prettier from 3.2.5 to 3.3.0 in /tests/wdio (#9952)
  web: bump chromedriver from 125.0.2 to 125.0.3 in /tests/wdio (#9953)
  core: groups: optimize recursive children query (#9931)
  core, web: update translations (#9938)
  stages/authenticator_webauthn: optimize device types creation (#9932)
  stages/authenticator_webauthn: Update FIDO MDS3 & Passkey aaguid blobs (#9936)
  providers/oauth2: don't handle api scope as special scope (#9910)
  core: bump goauthentik.io/api/v3 from 3.2024042.8 to 3.2024042.9 (#9915)
  web: bump the storybook group in /web with 7 updates (#9917)
  website/integrations: firezone: remove question mark typo and remove promotional sentence (#9927)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#9967)
  stages/captcha: rework (#9959)
  website/docs: fixed broken links to helm repo (#9960)
  core: bump pydantic from 2.7.2 to 2.7.3 (#9962)
  web: bump the wdio group in /tests/wdio with 4 updates (#9963)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (66 commits)
  rbac: filters: fix missing attribute for unauthenticated requests (#10061)
  tests/e2e: docker-compose.yml: remove version element forgotten last time (#10067)
  providers/microsoft_entra: fix error when updating connection attributes (#10039)
  website/integrations: aws: fix about service link (#10062)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in it (#10060)
  core: bump github.com/redis/go-redis/v9 from 9.5.2 to 9.5.3 (#10046)
  core: bump github.com/gorilla/websocket from 1.5.1 to 1.5.2 (#10047)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#10058)
  core: bump sentry-sdk from 2.5.0 to 2.5.1 (#10048)
  core: bump django-cte from 1.3.2 to 1.3.3 (#10049)
  core: bump packaging from 24.0 to 24.1 (#10050)
  web: bump yaml from 2.4.3 to 2.4.5 in /web (#10054)
  web: bump @sentry/browser from 8.7.0 to 8.8.0 in /web in the sentry group (#10051)
  web: bump esbuild from 0.21.4 to 0.21.5 in /web (#10053)
  crypto: update fingerprint at same time as certificate (#10036)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#10056)
  website/integrations: gitea: fix helm values (#10043)
  core, web: update translations (#10038)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#10040)
  docs/troubleshooting: upgrade docker: erroneous command (#10044)
  ...

kensternberg-authentik committed:
erge branch 'main' into dev

* main:
  website/developer-docs: add a baby Style Guide (#9900)
  website/integrations: gitlab: update certificate key pair location and specify sha (#9925)
  root: handle asgi exception (#10085)
  website: bump prettier from 3.3.1 to 3.3.2 in /website (#10082)
  web: bump prettier from 3.3.1 to 3.3.2 in /web (#10081)
  core: bump google-api-python-client from 2.132.0 to 2.133.0 (#10083)
  web: bump prettier from 3.3.1 to 3.3.2 in /tests/wdio (#10079)
  web: bump chromedriver from 125.0.3 to 126.0.0 in /tests/wdio (#10078)
  web: bump @sentry/browser from 8.8.0 to 8.9.1 in /web in the sentry group (#10080)
  web: bump braces from 3.0.2 to 3.0.3 in /web (#10077)
  website: bump braces from 3.0.2 to 3.0.3 in /website (#10076)
  web: bump braces from 3.0.2 to 3.0.3 in /tests/wdio (#10075)
  core: bump azure-identity from 1.16.0 to 1.16.1 (#10071)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: release notes for 2024.6 (#9812)
  policies/reputation: save to database directly (#10059)
  providers/enterprise: import user/group data when manually linking objects (#10089)
  core, web: update translations (#10108)
  web: Add enterprise / FIPS notification to the AdminOverviewPage (#10090)
  core: bump github.com/getsentry/sentry-go from 0.28.0 to 0.28.1 (#10095)
  web: bump API Client version (#10107)
  admin: system api: do not show FIPS status if no valid license (#10091)
  root: add configuration option to enable fips (#10088)
  web: bump the sentry group across 1 directory with 2 updates (#10101)
  web: bump ts-pattern from 5.1.2 to 5.2.0 in /web (#10098)
  web: bump the storybook group across 1 directory with 7 updates (#10102)
  core: bump github.com/gorilla/websocket from 1.5.2 to 1.5.3 (#10103)
  core: bump pydantic from 2.7.3 to 2.7.4 (#10093)
  core: bump bandit from 1.7.8 to 1.7.9 (#10094)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (27 commits)
  root: use custom model serializer that saves m2m without bulk (#10139)
  core: bump urllib3 from 2.2.1 to 2.2.2 (#10143)
  ci: bump docker/build-push-action from 5 to 6 (#10144)
  web: bump the storybook group in /web with 7 updates (#10147)
  web: bump ws from 8.16.0 to 8.17.1 in /web (#10149)
  web: fix needed because recent upgrade to task breaks spinner button (#10142)
  core: include version in built JS files (#9558)
  website/integations/services: Slack integration docs (#9933)
  web: fix early modal stack depletion (#10068)
  website: bump react-tooltip from 5.26.4 to 5.27.0 in /website (#10129)
  web: bump @patternfly/elements from 3.0.1 to 3.0.2 in /web (#10132)
  core: bump github.com/gorilla/sessions from 1.2.2 to 1.3.0 (#10135)
  web: bump chromedriver from 126.0.0 to 126.0.1 in /tests/wdio (#10136)
  core: bump github.com/spf13/cobra from 1.8.0 to 1.8.1 (#10133)
  core, web: update translations (#10127)
  core: bump ruff from 0.4.8 to 0.4.9 (#10128)
  core: bump goauthentik.io/api/v3 from 3.2024042.11 to 3.2024042.13 (#10134)
  core, web: update translations (#10118)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#10121)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#10120)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: update 2024.6 release notes with latest changes (#10174)
  core: fix error when raising SkipObject in mapping (#10153)
  website/docs: 2024.6 release notes: add note about group names (#10170)
  website/docs: update 2024.6 release notes with latest changes (#10167)
  core: bump twilio from 9.1.1 to 9.2.0 (#10162)
  web: bump chromedriver from 126.0.1 to 126.0.2 in /tests/wdio (#10161)
  web: bump the wdio group in /tests/wdio with 4 updates (#10160)
  website/integrations: jellyfin: add OIDC configuration (#9538)
  website/docs: Describe where to apply the auto setup env vars (#9863)
  website/integrations: gitlab: better service description (#9923)
  web: fix docker build for non-release versions (#10154)
  root: makefile: add codespell to make-website (#10116)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: Fix nginx proxy_pass directive documentation (#10181)
  website/docs: Remove hyphen in read replica in Release Notes (#10178)
  core: rework base for SkipObject exception to better support control flow exceptions (#10186)
  web: bump glob from 10.4.1 to 10.4.2 in /web (#10163)
  core: bump google-api-python-client from 2.133.0 to 2.134.0 (#10183)
  web: bump @sentry/browser from 8.9.2 to 8.10.0 in /web in the sentry group (#10185)
  website/docs: update template reference (#10166)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web/flows: update flow background (#10206)
  core: bump coverage from 7.5.3 to 7.5.4 (#10201)
  core: bump twilio from 9.2.0 to 9.2.1 (#10202)
  web: bump chromedriver from 126.0.2 to 126.0.3 in /tests/wdio (#10203)
  web: bump @sentry/browser from 8.10.0 to 8.11.0 in /web in the sentry group (#10204)
  website: bump typescript from 5.4.5 to 5.5.2 in /website (#10190)
  web: bump typescript from 5.4.5 to 5.5.2 in /web (#10191)
  web: bump typescript from 5.4.5 to 5.5.2 in /tests/wdio (#10192)
  core: bump ruff from 0.4.9 to 0.4.10 (#10193)
  core: bump selenium from 4.21.0 to 4.22.0 (#10194)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (52 commits)
  web: provide a test framework (#9681)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#10272)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#10271)
  core: bump drf-jsonschema-serializer from 2.0.0 to 3.0.0 (#10262)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in ru (#10268)
  core: bump goauthentik.io/api/v3 from 3.2024042.13 to 3.2024060.1 (#10260)
  core, web: update translations (#10259)
  website/docs: update geoip and asn documentation following field changes (#10265)
  web: provide better feedback on Application Library page about search results (#9386)
  web: disable reading dark mode out of the UI by default (#10256)
  web/flows: remove continue button from AutoSubmit stage (#10253)
  web: bump API Client version (#10252)
  website/docs: update geoip and asn example to use the proper syntax (cherry-pick #10249) (#10250)
  website/docs: update the Welcome page (#10222)
  website/docs: update geoip and asn example to use the proper syntax (#10249)
  security: update supported versions (cherry-pick #10247) (#10248)
  security: update supported versions (#10247)
  website/docs: remove RC disclaimer from 2024.6 release notes (cherry-pick #10245) (#10246)
  website/docs: remove RC disclaimer from 2024.6 release notes (#10245)
  website/docs: update 2024.6 release notes with latest changes (cherry-pick #10228) (#10243)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: lint package-lock.json file (#10157)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: bump API Client version (#10290)
  web/admin: show matching user reputation scores in user details (#10276)
  tests/e2e: fix ldap tests following #10270 (#10288)
  root: fix web docker build (#10287)
  core: fix URLValidator regex to allow single digit port (#10280)
  root: allow extra sentry settings (#10269)
  core, web: update translations (#10279)
  core: bump sentry-sdk from 2.5.1 to 2.7.1 (#10282)
  core: bump google-api-python-client from 2.134.0 to 2.135.0 (#10281)
  core: bump twilio from 9.2.1 to 9.2.2 (#10283)
  core: bump swagger-spec-validator from 3.0.3 to 3.0.4 (#10284)
  core: bump ruff from 0.4.10 to 0.5.0 (#10285)
  web: bump @sentry/browser from 8.12.0 to 8.13.0 in /web in the sentry group (#10286)
  core: adjust styling to meet our standards (#10277)
  events: associate login_failed events to a user if possible (#10270)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  core: applications api: prefetch related policies (#10273)
  web: restore hasLaunchUrl to client-side criteria for filtering apps (#10291)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: fix package lock out of sync (#10314)
  core, web: update translations (#10294)
  stages/authenticator_webauthn: Update FIDO MDS3 & Passkey aaguid blobs (#10306)
  web: bump the esbuild group in /web with 2 updates (#10308)
  web: bump esbuild from 0.21.5 to 0.22.0 in /web (#10309)
  website: bump postcss from 8.4.38 to 8.4.39 in /website (#10310)
  core: bump goauthentik.io/api/v3 from 3.2024060.1 to 3.2024060.2 (#10311)
  core: bump psycopg from 3.1.19 to 3.2.1 (#10313)
  stages/user_login: fix ?next parameter not carried through broken session binding (#10301)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  web: provide default endpoint api configuration (#10319)
  provider/scim: Fix exception handling for missing ServiceProviderConfig (#10322)
  translate: Updates for file web/xliff/en.xlf in fr (#10334)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#10293)
  web: bump esbuild from 0.22.0 to 0.23.0 in /web (#10331)
  web: fixed missed internationalized strings (#10323)
  core, web: update translations (#10324)
  website: bump typescript from 5.5.2 to 5.5.3 in /website (#10326)
  core: bump pydantic from 2.7.4 to 2.8.0 (#10325)
  web: bump typescript from 5.5.2 to 5.5.3 in /tests/wdio (#10327)
  web: bump the esbuild group in /web with 2 updates (#10330)
  web: bump typescript from 5.5.2 to 5.5.3 in /web (#10332)
  web: set noopener and noreferrer on all external links (#10304)
  core: remove transitionary old JS urls (#10317)
  web/flows: remove background image link (#10318)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#10315)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  website/docs: add links and tweaks to existing docs on flow executors (#10340)
  sources/saml: fix pickle error, add saml auth tests (#10348)
  web: bump API Client version (#10351)
  core: applications api: add option to only list apps with launch url (#10336)
  website/integrations: minio: configure openid on web (#9874)
  website/docs: integrations: gitea: specify callback url (#10180)
  providers/saml: fix metadata import error handling (#10349)
  core, web: update translations (#10341)
  core: bump twilio from 9.2.2 to 9.2.3 (#10343)
  core: bump google-api-python-client from 2.135.0 to 2.136.0 (#10344)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#10339)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#10338)
  web: bump the storybook group in /web with 7 updates (#10263)
  web: lintpicking (#10212)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  docs/troubleshooting: upgrade docker: prompt user to ensure pg_dump'd file has data (#10353)

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (29 commits)
  website/integrations: aws: cleanup (#10355)
  web: bump API Client version (#10389)
  web/flows: Simplified flow executor (#10296)
  website/docs: sources: ldap: remove extra example (#10387)
  website/docs: add new content from old PR #9524 (#10158)
  stages/authenticator_validate: fix friendly_name being required (#10382)
  core: bump go api client (#10383)
  web: bump API Client version (#10381)
  outposts: make refresh interval configurable (#10138)
  core, web: update translations (#10378)
  web: bump @sentry/browser from 8.13.0 to 8.14.0 in /web in the sentry group (#10379)
  core: bump goauthentik.io/api/v3 from 3.2024060.3 to 3.2024060.4 (#10380)
  sources/oauth: fix link not being saved (#10374)
  website/docs: update postgres on docker: fix backtick (#10372)
  website/integrations: apache guacamole: cleanup doc page (#10354)
  web: bump API Client version (#10371)
  Revert "core: applications api: add option to only list apps with launch url (#10336)" (#10370)
  web: bump @wdio/cli from 8.39.0 to 8.39.1 in /web (#10362)
  core: bump goauthentik.io/api/v3 from 3.2024060.2 to 3.2024060.3 (#10356)
  website: bump react-tooltip from 5.27.0 to 5.27.1 in /website (#10357)
  ...

kensternberg-authentik committed:
Merge branch 'main' into dev

* main:
  api: use custom json renderer for speed (#9977)
  core: bump github.com/sethvargo/go-envconfig from 1.0.3 to 1.1.0 (#10405)
  core: bump goauthentik.io/api/v3 from 3.2024060.5 to 3.2024060.6 (#10404)
  web: bump glob from 10.4.2 to 10.4.3 in /web (#10407)
  web: bump @swc/core from 1.6.7 to 1.6.13 in /web/sfe (#10408)
  core: bump ruff from 0.5.0 to 0.5.1 (#10390)
  web: bump @swc/cli from 0.3.14 to 0.4.0 in /web/sfe (#10394)
  core: bump certifi from 2024.2.2 to 2024.7.4 (#10398)
  web: bump @swc/core from 1.6.6 to 1.6.7 in /web/sfe (#10395)
  web: bump @sentry/browser from 8.14.0 to 8.15.0 in /web in the sentry group across 1 directory (#10388)

rissson committed:
root: fix migrations missing using db_alias

Signed-off-by: Marc 'risson' Schmitt <marc.schmitt@risson.space>

rissson committed:
more

Signed-off-by: Marc 'risson' Schmitt <marc.schmitt@risson.space>

kensternberg-authentik committed:
Merge remote-tracking branch 'origin/migrations-fix-missing-using' into dev

* origin/migrations-fix-missing-using:
  more
  root: fix migrations missing using db_alias

kensternberg-authentik committed:
Merge branch 'main' into dev

* main: (35 commits)
  core: bump github.com/redis/go-redis/v9 from 9.5.3 to 9.5.4 (#10469)
  core: bump goauthentik.io/api/v3 from 3.2024060.6 to 3.2024061.1 (#10470)
  web: bump the babel group across 1 directory with 2 updates (#10471)
  web: bump the storybook group across 1 directory with 7 updates (#10472)
  core: bump coverage from 7.5.4 to 7.6.0 (#10473)
  website/docs: air gapped: clarify .env usage at the top for Kubernetes installations (#10447)
  website/docs: air gapped: update "see configuration" wording (#10448)
  website/docs: Add Kubernetes Bootstrap Instructions (#9541)
  web: bump API Client version (#10461)
  release: 2024.6.1
  website/docs: add 2024.6.1 release notes (#10456)
  core: revert backchannel only filtering (#10455)
  website/docs: add info about license:org relation (#10444)
  web: bump @sentry/browser from 8.15.0 to 8.17.0 in /web in the sentry group across 1 directory (#10449)
  web: bump the storybook group across 1 directory with 7 updates (#10451)
  core: bump msgraph-sdk from 1.4.0 to 1.5.2 (#10450)
  core: bump sentry-sdk from 2.8.0 to 2.9.0 (#10452)
  website/docs: add info and link about supported versions (#10442)
  core: fix source flow_manager not resuming flow when linking (#10436)
  ci: fix PR comment k8s snippet (#10433)
  ...

kensternberg-authentik committed:
website: add links to schema references

Added links to the schema references on the blueprints page, and a new collapsible
section that shows all the past schemas as referenced in the original edit.

URLs to live objects (URLs that aren't just examples) should be
live links themselves.  Referring to past versions of the documents
and not providing a list of those versions or links to them seemed
incomplete.

BeryJu commented on website/developer-docs/blueprints/v1/structure.md:
```diff
@@ -1,10 +1,34 @@
+[`https://version-2023-4.goauthentik.io/blueprints/schema.json`](https://version-2023-4.goauthentik.io/blueprints/schema.json).
+
+<Collapse title="Available Older Blueprint Schemas">
+- [Version 2022-8](https://version-2022-8.goauthentik.io/blueprints/schema.json)
```
We really need to have some sort of automation to generate a list of versions instead of hardcoding this, I can already for see this getting outdated (and also maybe order in the reverse of this starting with current and getting older?)

kensternberg-authentik committed:
Merge branch 'main' into website/edit/links-to-schemas

* main: (867 commits)
  website: bump elliptic from 6.5.7 to 6.6.0 in /website (#11869)
  core: bump selenium from 4.25.0 to 4.26.0 (#11875)
  core: bump goauthentik.io/api/v3 from 3.2024083.14 to 3.2024100.1 (#11876)
  website/docs: add info about invalidation flow, default flows in general (#11800)
  website: fix docs redirect (#11873)
  website: remove RC disclaimer for version 2024.10 (#11871)
  website: update supported versions (#11841)
  web: bump API Client version (#11870)
  root: backport version bump 2024.10.0 (#11868)
  website/docs: 2024.8.4 release notes (#11862)
  web/admin: provide default invalidation flows for LDAP and Radius (#11861)
  core, web: update translations (#11858)
  web/admin: fix code-based MFA toggle not working in wizard (#11854)
  sources/kerberos: add kiprop to ignored system principals (#11852)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#11846)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in it (#11845)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#11847)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#11848)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#11849)
  translate: Updates for file web/xliff/en.xlf in it (#11850)
  ...

kensternberg-authentik committed:
website: add a script to automatically replace the list of available schemas.
This script, `list-blueprint-schemas`, is a blunt and fragile object, but it can get the work done.  Every time
it is run, it will get a list of all version tags from Git, compare them to all the available schemas available
by version, and print out a copy of `./website/docs/customize/blueprints/v1/structure.md` where it has replaced
the existing `Available Older Blueprint Schemas` collapse block with the most up-to-date version.

I will have to talk to @beryju and @tanberry to find out what they want done with this; would it be fine to just
run this at build time *before* the site is compiled, replacing the structure.md file with this dynamic block,
or can we generate it separately and [include it as a markdown
extension](https://stackoverflow.com/questions/65741307/how-to-include-import-markdown-file-content-in-docusaurus-v2)?

BeryJu committed:
use dots for version numbers

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
run as part of build

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
format

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
write file back

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

kensternberg-authentik commented:
Nobody's quite sure what to make of this. I'm going to close it until we have a need for it.
